imagebuildah/cache/local_layer_provider.go

```diff
@@ -0,0 +1,135 @@
+package cache
+
+import (
+	"context"
+	"sync"
+
+	"github.com/containers/image/v5/manifest"
+	is "github.com/containers/image/v5/storage"
+	"github.com/containers/storage"
+	"github.com/opencontainers/go-digest"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// LocalLayerProvider is an implementation of the LayerProvider that
+type LocalLayerProvider struct {
+	store storage.Store
+
+	localLayers     map[string]string
+	localLayersLock sync.Mutex
+}
+
+// NewLocalLayerProvider creates a new instance of the LocalLayerProvider.
+func NewLocalLayerProvider(store storage.Store) LocalLayerProvider {
```

Comment:
This looks like it's supposed to be returning either an instance of the `LayerProvider` interface.  If not that, then a pointer to this structure because passing or returning structures that contain mutexes is generally a source of bugs, even though we know that this specific instance doesn't create a problem.

Reply:
I extended the interface with PopulateLayer and now I am returning it.

II was thinking to keep the PopulateLayer out of it because this is exceptional only for the LocalLayerProvider, but I guess the other providers can have it implemented - noop.